libkbfs/bare_root_metadata.go

```diff
@@ -562,40 +565,67 @@ func (md *BareRootMetadataV2) IsValidAndSigned(
 	}
 
 	if (md.MergedStatus() == Merged) != (md.BID() == NullBranchID) {
-		return errors.New("Branch ID doesn't match merged status")
+		return fmt.Errorf("Branch ID %s doesn't match merged status %s",
+			md.BID(), md.MergedStatus())
 	}
 
 	handle, err := md.MakeBareTlfHandle()
 	if err != nil {
 		return err
 	}
 
-	writer := md.LastModifyingWriter()
-
 	// Make sure the last writer is valid.
+	writer := md.LastModifyingWriter()
 	if !handle.IsWriter(writer) {
-		return errors.New("Invalid modifying writer")
+		return fmt.Errorf("Invalid modifying writer %s", writer)
+	}
+
+	// Make sure the last modifier is valid.
+	user := md.LastModifyingUser
+	if !handle.IsReader(user) {
+		return fmt.Errorf("Invalid modifying user %s", user)
+	}
+
+	// Verify signature. We have to re-marshal the WriterMetadata,
+	// since it's embedded.
+	buf, err := codec.Encode(md.WriterMetadataV2)
+	if err != nil {
+		return err
+	}
+
+	err = crypto.Verify(buf, md.WriterMetadataSigInfo)
+	if err != nil {
+		return fmt.Errorf("Could not verify writer metadata: %v", err)
 	}
 
+	return nil
+}
+
+// IsLastModifiedBy implements the BareRootMetadata interface for
+// BareRootMetadataV2.
+func (md *BareRootMetadataV2) IsLastModifiedBy(
+	currentUID keybase1.UID, currentVerifyingKey VerifyingKey) error {
```

Comment:
I don't think it makes sense to call these parameters "current" anymore.  This function doesn't care whether the values passed in belong to the current user or not, it's just checking whether some user+key matches the last modifier.